src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -613,26 +629,49 @@ private void processLogsAndUpload() throws InterruptedException {
                             if (componentCurrentProcessingLogFile.containsKey(componentName)) {
                                 CurrentProcessingFileInformation processingFileInformation =
                                         componentCurrentProcessingLogFile.get(componentName);
-                                if (processingFileInformation.fileName.equals(file.getAbsolutePath())
-                                        && processingFileInformation.lastModifiedTime == file.lastModified()) {
+                                //if (processingFileInformation.fileName.equals(file.getAbsolutePath())
+                                if (processingFileInformation.fileHash.equals(fileHash)
+                                        && processingFileInformation.lastModifiedTime <= file.lastModified()) {
                                     startPosition = processingFileInformation.startPosition;
                                 }
                             }
-                            LogFileInformation logFileInformation = LogFileInformation.builder()
-                                            .logFile(file)
-                                            .startPosition(startPosition)
-                                            .fileHash(fileHash)
-                                            .build();
+
+                            // If the file is not completed read or the file is recorded as
+                            // update the global map of hash and name pair
+                            fileHashNamePairs.putIfAbsent(fileHash, file.getAbsolutePath());
+                            if (!fileHashNamePairs.get(fileHash).equals(file.getAbsolutePath())) {
```

Comment:
Seems unnecessary - We are saying if the absolute path has changed for this file has that I have stored. Update the value. Why not just always update it?